src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -228,7 +228,28 @@ module.exports = function(
   }
 
 
-  function markUpdate(workInProgress) {
+  function markUpdateIfNecessary(workInProgress) {
+    const current = workInProgress.alternate;
+    const instance = workInProgress.stateNode;
+    let shouldMarkUpdate = false;
+    if (current) {
+      // During an update, we might either need to detach a ref
+      // or to call the componentDidUpdate() hook in the commit phase.
+      const currentRef = current.ref;
```

Comment:
Or wait, should ref change trigger a `componentDidUpdate` call? I assumed no but maybe it should.